### PR TITLE
N2O reaction tweaks so people don't surprise blow up

### PR DESCRIPTION
**Pull request by Thalpy**
## About The Pull Request

N2O at the moment is very very tight on it's explosive window, so it has a high chance to blow up a chemist who isn't ready to crash their temp as soon as it starts. This reduces the temperature spike so it's less dangerous and will always resolve without an explosion if the temperature is crept up. In addition; if the reaction has 30% or less purity, part of the N2O will convert to gas.

## Why It's Good For The Game

Reactions involved as components to others should not have extreme lethality.

## Changelog
:cl:
fix: fixes N2O to make it less difficult to react.
tweak: N2O will now produce N2O gas when below 30% purity during a reaction.
/:cl:

### CODE REVIEW

**Comment by Jared-Fogle** [comment]
> In addition; if the reaction has 30% or less purity, part of the N2O will convert to gas.

Should be in the changelog.

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 1]
```diff
@@ -123,6 +123,24 @@
...
+/datum/chemical_reaction/nitrous_oxide/overly_impure(datum/reagents/holder, datum/equilibrium/equilibrium)
+	. = ..()
+	var/turf/exposed_turf = get_turf(holder.my_atom)
+	if(!exposed_turf)
+		return
+	exposed_turf.atmos_spawn_air("n2o=[equilibrium.step_target_vol/2];TEMP=[holder.chem_temp]")
+	clear_products(holder, equilibrium.step_target_vol)
+
+/datum/chemical_reaction/nitrous_oxide/overheated(datum/reagents/holder, datum/equilibrium/equilibrium)
+	return //When I refactor explosions, this will be used, but for now, the alternative reaction will catch the high temperature, so we don't need to do anything. This is so that on lookup, the overheat value will be displayed.
```
Comment (Jared-Fogle) [inline code/modules/reagents/chemistry/recipes/others.dm:142]: This comment seems more like a note to yourself than something that's useful for other developers.

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 2]
```diff
@@ -123,6 +123,24 @@
...
+	purity_min = 0.3
+	thermic_constant = 35 //gives a bonus 15C wiggle room
+	rate_up_lim = 25 //Give a chance to pull back
+	
+/datum/chemical_reaction/nitrous_oxide/overly_impure(datum/reagents/holder, datum/equilibrium/equilibrium)
+	. = ..()
+	var/turf/exposed_turf = get_turf(holder.my_atom)
+	if(!exposed_turf)
+		return
+	exposed_turf.atmos_spawn_air("n2o=[equilibrium.step_target_vol/2];TEMP=[holder.chem_temp]")
```
Comment (Jared-Fogle) [inline code/modules/reagents/chemistry/recipes/others.dm:138]: @LemonInTheDark Shouldn't be any problems with this, yes?

File: code/modules/reagents/chemistry/recipes/others.dm (hunk 1 above)
Comment (Thalpy) [reply code/modules/reagents/chemistry/recipes/others.dm:142]: It's more to explain the pointless overheat proc, but I'm happy with removing it.

Comment (Jared-Fogle) [reply code/modules/reagents/chemistry/recipes/others.dm:142]: Can you find a way to explain it in a way that doesn't mention future refactors? As in, document why this proc is empty *now*.

Comment (Thalpy) [reply code/modules/reagents/chemistry/recipes/others.dm]: Will do!

File: code/modules/reagents/chemistry/recipes/others.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/modules/reagents/chemistry/recipes/others.dm:138]: Yeah that'll do

**Review by Mothblocks — APPROVED** [review]
Fix conflicts, and it's good.

**Pull request merged**